artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -231,6 +232,9 @@ public void closed() {
    public void disconnect(ServerConsumer consumerId, String errorDescription) {
       StompSubscription stompSubscription = subscriptions.remove(consumerId.getID());
       if (stompSubscription != null) {
+         if (stompSubscription.isNoLocal()) {
+            noLocalSubscriptionCount.decrementAndGet();
+         }
```

Comment:
This disconnect method looks to be the only place a decrement is done...what if the subscription is explicitly unsubscribed and so this doesnt occur?

Reply:
Fixed.